assayist/processor/utils.py

```diff
@@ -84,7 +84,40 @@ def download_build(build_identifier, output_dir):
                 artifacts.append(file_path)
                 log.info(f'Downloaded {os.path.split(file_path)[-1]}')
 
-    return artifacts
+    return artifacts, build
+
+
+def download_source(build_info, output_dir):
+    """
+    Download the source (from dist-git) that was used in the specified build.
+
+    :param build_info: build information from koji.getBuild()
+    :param output_dir: the path to download the source to
+    """
+    # Make sure the git command is installed
+    _assert_command('git')
+
+    source_url = build_info.get('source')
+    if not source_url:
+        raise RuntimeError(f'Build {build_info["id"]} has no associated source URL.')
+
+    log.info(f'Downloading source for {build_info["id"]}')
+
+    url, _, commit_id = source_url.partition('#')
+    component = url.split('/')[-1]
+
+    cmd = ['git', 'clone', url]
+    process = subprocess.Popen(cmd, cwd=output_dir,
+                               stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+
+    # For some reason, any errors are streamed to stdout and not stderr
+    output, _ = process.communicate()
+    output = output.decode('utf-8')
+    if process.returncode != 0:
+        raise RuntimeError(f'The command "{" ".join(cmd)}" failed with: {output}')
+
+    cmd = ['git', 'reset', '--hard', commit_id]
+    subprocess.Popen(cmd, cwd=os.path.join(output_dir, component), stdout=subprocess.PIPE)
```

Comment:
You should do some error checking here

Reply:
Is there anything that could fail in this command if the previous `git clone` succeeded? I suppose if the `commit_id` did not exist this could fail, is it worth handling that?

Reply:
In general I think it's just better to do error checking, but if you think it's unnecessary, I won't block on it.